File: Numerical_Methods/Solving Systems of Linear Equations/Gauss-Seidel Iterations/python_solutions/problem4.py
def gauss_seidel():
    # System of equations:
    # 4x + y - z = 5
    # x + 3y + z = 6
    # -x + y + 4z = 7
    
    x = y = z = 1.0  # Initial values
    tolerance = 0.001
    iteration = 0
    converged = False

    print("Gauss-Seidel Iteration Method - Problem 4")
    print(f"Initial values: x = {x}, y = {y}, z = {z}")
    print(f"Tolerance: {tolerance}")
    print("\nIterations:")

    while not converged:
        iteration += 1
        
        # Calculate new values using Gauss-Seidel formula
        x_new = (5 - y + z) / 4
        y_new = (6 - x_new - z) / 3
        z_new = (7 + x_new - y_new) / 4

        # Check for convergence
        if (abs(x_new - x) < tolerance and
            abs(y_new - y) < tolerance and
            abs(z_new - z) < tolerance):
            converged = True

        # Update values
        x, y, z = x_new, y_new, z_new

        # Print current iteration
        print(f"Iteration {iteration}: x = {x:.6f}, y = {y:.6f}, z = {z:.6f}")

    print(f"\nSolution converged after {iteration} iterations:")
    print(f"x = {x:.6f}")
    print(f"y = {y:.6f}")
    print(f"z = {z:.6f}")

if __name__ == "__main__":
    gauss_seidel() 

File: Numerical_Methods/Solving Systems of Linear Equations/Gauss-Seidel Iterations/java_solutions/Problem4.java
public class Problem4 {
    public static void main(String[] args) {
        // System of equations:
        // 4x + y - z = 5
        // x + 3y + z = 6
        // -x + y + 4z = 7
        
        double x = 1.0, y = 1.0, z = 1.0; // Initial values
        double xNew, yNew, zNew;
        double tolerance = 0.001;
        int iteration = 0;
        boolean converged = false;

        System.out.println("Gauss-Seidel Iteration Method - Problem 4");
        System.out.println("Initial values: x = " + x + ", y = " + y + ", z = " + z);
        System.out.println("Tolerance: " + tolerance);
        System.out.println("\nIterations:");

        while (!converged) {
            iteration++;
            
            // Calculate new values using Gauss-Seidel formula
            xNew = (5 - y + z) / 4;
            yNew = (6 - xNew - z) / 3;
            zNew = (7 + xNew - yNew) / 4;

            // Check for convergence
            if (Math.abs(xNew - x) < tolerance &&
                Math.abs(yNew - y) < tolerance &&
                Math.abs(zNew - z) < tolerance) {
                converged = true;
            }

            // Update values
            x = xNew;
            y = yNew;
            z = zNew;

            // Print current iteration
            System.out.printf("Iteration %d: x = %.6f, y = %.6f, z = %.6f%n", 
                            iteration, x, y, z);
        }

        System.out.println("\nSolution converged after " + iteration + " iterations:");
        System.out.printf("x = %.6f%n", x);
        System.out.printf("y = %.6f%n", y);
        System.out.printf("z = %.6f%n", z);
    }
} 